test/test_decomp.py

```diff
@@ -159,6 +159,8 @@ def op_assert_ref(test_case, op, test_dtype, i, orig, decomp, ref, args, kwargs)
         (torch.float16, torch.ops.aten.native_batch_norm.default): 1e-5,
         (torch.bfloat16, torch.ops.aten.linalg_vector_norm.default): 1e-6,
         (torch.float16, torch.ops.aten.linalg_vector_norm.default): 1e-6,
+        (torch.float16, torch.ops.aten.glu.default): 1e-2,
+        (torch.bfloat16, torch.ops.aten.glu.default): 1e-2,
```

Comment:
One thing I wanted to check explicitly is this. The tests don't pass unless I make the precision this low. It's only for {b,}float16 so maybe it's okay - what do you think @ngimel ?

Reply:
fwiw, this comes from the fact that the sigmoid is computed internally without casting yhe intermediate results to float32 as we do in TI

Reply:
Oh you are right @Lezcano, we should fix that in the eager implementation

Reply:
I think eager is fine here, we should fix the ref. 
In the ref, we should first cast the inputs to float32, perform the computations in float32, and then cast the result back to whatever lower precision type we had.

Note that this behaviour is shared by all the TI implementations in eager, so this could very well be implemented as a decorator.

Reply:
Just add type promotion wrapper here as was already suggested https://github.com/pytorch/pytorch/pull/82214#discussion_r939167379 and tests will pass without tolerance overrides.
```py
@elementwise_type_promotion_wrapper(
    type_promoting_args=("a",),
    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
)
```

Reply:
ag, that's it, thank you :)

Reply:
Yes, thanks!

Reply:
Eager doesn't do type promotion to opmath https://github.com/pytorch/pytorch/blob/c3e67d8a8c1217c6c65c0539866e6c59ee2bd9a4/aten/src/ATen/native/cuda/UnarySpecialOpsKernel.cu#L154, whereas sigmoid ref does (all unary refs get wrapped in type promotion wrapper by default). glu ref also needs type promotion wrapper.

Reply:
We looked at it the other day and I completely missed that one! Thank you @ngimel!

Reply:
Thansk guys. @ngimel I think this should be ready to go, could you PTAL?

Reply:
Discussed it with @Lezcano and I will also fix up cpu and cuda implementations of sigmoid to do the type promotion for {,b}float16